### ISSUE TITLE
Improve branch name ticket detection

### ISSUE THREAD

**Issue by tombruijn** [tombruijn/lintje#25]
Given this example:

```
BranchNameTicketNumber: A ticket number was detected in the branch name
  Branch:1: elixir-1.13.2-ci
  |
  | elixir-1.13.2-ci
  | ^^^^^^^^^^^^^^^^ Remove the ticket number from the branch name or expand the branch name with more details
```

- [x] It should only fail on `elixir-12`, not `elixir-1`. Update the rule to expect at least 2 digits.
- [x] Check if the branch name contains dots or dashes, that are followed by other digits. That's probably a version number and can be considered a proper branch name.
